### PR TITLE
Rdm 5303 pr

### PR DESCRIPTION
**Pull request by Thor-tech-of-metal**
RDM-5303: Document file name disappears when you come back to the page from a previous page.
    
      [RDM-5303](https://tools.hmcts.net/jira/browse/RDM-5303).

### CODE REVIEW

File: src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts [hunk 1]
```diff
@@ -118,6 +118,13 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
     return !this.pageValidationService.isPageValid(this.currentPage, this.editForm);
   }
 
+  toPreviousPage() {
+    let pageFormFields = this.formValueService.filterCurrentPageFields(this.currentPage.case_fields, this.editForm.value);
+    let caseEventData: CaseEventData = this.formValueService.sanitise(pageFormFields) as CaseEventData;
+    this.updateFormData(caseEventData as CaseEventData);
+    this.previous();
+  }
+
```
Comment (smathangi) [inline src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts]: may be a good unit test to cover this logic?

Comment (Thor-tech-of-metal) [reply src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts]: I can add another test that evaluate the call to comp.updateFormData(jsonData);  and this.previous ()

File: src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts [hunk 2]
```diff
@@ -118,6 +118,13 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
     return !this.pageValidationService.isPageValid(this.currentPage, this.editForm);
   }
 
+  toPreviousPage() {
+    let pageFormFields = this.formValueService.filterCurrentPageFields(this.currentPage.case_fields, this.editForm.value);
+    let caseEventData: CaseEventData = this.formValueService.sanitise(pageFormFields) as CaseEventData;
+    this.updateFormData(caseEventData as CaseEventData);
+    this.previous();
```
Comment (Thor-tech-of-metal) [inline src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts]: It seems that there are some steps that should be added sanitiseDynamicLists and caseEventData.event_data = this.editForm.value.data; others I will check it

Comment (Thor-tech-of-metal) [reply src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts]: DONE

File: src/shared/components/case-editor/services/cases.service.ts [hunk 3]
```diff
@@ -22,7 +22,8 @@ export class CasesService {
     'application/vnd.uk.gov.hmcts.ccd-data-store-api.ui-start-draft-trigger.v2+json;charset=UTF-8';
 
   // External (Data Store) API
-  public static readonly V2_MEDIATYPE_CASE_DOCUMENTS = 'application/vnd.uk.gov.hmcts.ccd-data-store-api.case-documents.v2+json;charset=UTF-8';
+  public static readonly V2_MEDIATYPE_CASE_DOCUMENTS = 'application/vnd.uk.gov.' +
```
Comment (mario-paniccia) [inline src/shared/components/case-editor/services/cases.service.ts]: can you revert this change? if line is too long you could use this formatting: 
```
public static readonly V2_MEDIATYPE_CASE_DATA_VALIDATE =
    'application/vnd.uk.gov.hmcts.ccd-data-store-api.case-data-validate.v2+json;charset=UTF-8'
```

Comment (Thor-tech-of-metal) [reply src/shared/components/case-editor/services/cases.service.ts]: yep done :)

File: src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts [hunk 4]
```diff
@@ -118,6 +118,21 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
...
+    let pageFormFields = this.formValueService.filterCurrentPageFields(this.currentPage.case_fields, this.editForm.value);
+    this.formValueService.sanitiseDynamicLists(this.currentPage.case_fields, pageFormFields);
+    let caseEventData: CaseEventData = this.formValueService.sanitise(pageFormFields) as CaseEventData;
+    caseEventData.event_token = this.eventTrigger.event_token;
+    caseEventData.ignore_warning = this.ignoreWarning;
+    caseEventData.event_data = this.editForm.value.data;
+
+    if (this.caseEdit.caseDetails) {
+      caseEventData.case_reference = this.caseEdit.caseDetails.case_id;
+    }
```
Comment (mario-paniccia) [inline src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts]: this is duplicated of the code in `submit` method. We should removed this duplication.

File: src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts [hunk 5]
```diff
@@ -118,6 +118,21 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
     return !this.pageValidationService.isPageValid(this.currentPage, this.editForm);
   }
 
+  toPreviousPage() {
```
Comment (mario-paniccia) [inline src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts]: this seems like a big change in the behaviour of the previous method. I wonder whether the current behaviour was like that by design. 
What we can do is we can approve these changes but ask QAs to do full regression testing on the behaviour of navigating to previous pages.

File: src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts (hunk 4 above)
Comment (Thor-tech-of-metal) [reply src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts]: Mario  I am planning to create a generic method for this only.  It will be re-used  for submit() and  toPreviousPage() . Is it OK?

let pageFormFields = this.formValueService.filterCurrentPageFields(this.currentPage.case_fields, this.editForm.value);
    this.formValueService.sanitiseDynamicLists(this.currentPage.case_fields, pageFormFields);
    let caseEventData: CaseEventData = this.formValueService.sanitise(pageFormFields) as CaseEventData;
    caseEventData.event_token = this.eventTrigger.event_token;
    caseEventData.ignore_warning = this.ignoreWarning;
    caseEventData.event_data = this.editForm.value.data;

    if (this.caseEdit.caseDetails) {
      caseEventData.case_reference = this.caseEdit.caseDetails.case_id;
    }

File: src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts [hunk 6]
```diff
@@ -118,21 +118,29 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
...
+    caseEventData.ignore_warning = this.ignoreWarning;
+    caseEventData.event_data = this.editForm.value.data;
+    if (this.caseEdit.caseDetails) {
+      caseEventData.case_reference = this.caseEdit.caseDetails.case_id;
+    }
+    return caseEventData;
+  }
+  toPreviousPage() {
+    let caseEventData: CaseEventData = this.buildCaseEventData();
+    this.updateFormData(caseEventData as CaseEventData);
```
Comment (mario-paniccia) [inline src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts]: why do you need `as CaseEventData` here?

File: src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts [hunk 7]
```diff
@@ -118,21 +118,29 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
...
+    this.formValueService.sanitiseDynamicLists(this.currentPage.case_fields, pageFormFields);
+    let caseEventData: CaseEventData = this.formValueService.sanitise(pageFormFields) as CaseEventData;
+    caseEventData.event_token = this.eventTrigger.event_token;
+    caseEventData.ignore_warning = this.ignoreWarning;
+    caseEventData.event_data = this.editForm.value.data;
+    if (this.caseEdit.caseDetails) {
+      caseEventData.case_reference = this.caseEdit.caseDetails.case_id;
+    }
+    return caseEventData;
+  }
```
Comment (mario-paniccia) [inline src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts]: add an empty line between methods

File: src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts [hunk 8]
```diff
@@ -118,21 +118,29 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
     return !this.pageValidationService.isPageValid(this.currentPage, this.editForm);
   }
 
+   buildCaseEventData(): CaseEventData {
```
Comment (mario-paniccia) [inline src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts]: make it private and put it below all public methods at the bottom of the file
